AutoFlow_OmicsDataHandling/INCA_script_generator.py

```diff
@@ -0,0 +1,1232 @@
+                if model_rxn_id == "Ec_Biomass_INCA":
+                    rxn_equation += a
+                    continue
+                else:
+                    # we save the location of the row corresponding to
+                    # this reaction in the other file
+                    # "prepare_input" is a helper function to make the
+                    # entries useable, this will have to be checked with
+                    # other input
+
+                    # Atom Mapping
+                    index_in_atomMapping = list(
+                        atomMappingReactions_data_I["rxn_id"]
+                    ).index(modelReaction_data["rxn_id"])
+                    atomMapping_reactants_stoichiometry = prepare_input(
+                        atomMappingReactions_data_I.iloc[
+                            index_in_atomMapping
+                        ].loc["reactants_stoichiometry_tracked"],
+                        "Curly",
+                    )
+                    atomMapping_products_stoichiometry = prepare_input(
+                        atomMappingReactions_data_I.iloc[
+                            index_in_atomMapping
+                        ].loc["products_stoichiometry_tracked"],
+                        "Curly",
+                    )
+
+                    # this step makes sure that we exclude reactions that
+                    # do not have any stoichiometry annotated
+                    if (
+                        atomMapping_reactants_stoichiometry[0] == ""
+                        and atomMapping_products_stoichiometry[0] == ""
+                    ):
+                        print(
+                            "There is no stoichimetriy given for:",
+                            model_rxn_id,
+                        )
+                        # Model
+                        model_reactants_ids = prepare_input(
+                            modelReaction_data_I.iloc[index_in_model].loc[
+                                "reactants_ids"
+                            ],
+                            "Curly",
+                        )
+                        model_products_ids = prepare_input(
+                            modelReaction_data_I.iloc[index_in_model].loc[
+                                "products_ids"
+                            ],
+                            "Curly",
+                        )
+                        model_reactants_stoichiometry = [
+                            float(i) * -1
+                            for i in model_reactants_stoichiometry
+                        ]
+                        model_products_stoichiometry = [
+                            float(i) for i in model_products_stoichiometry
+                        ]
+
+                        for reactant_cnt, reactant in enumerate(
+                            model_reactants_ids
+                        ):
+                            # this little bit adds +s if it's not
```

Comment:
Were you having issues with duplicate reactions?  There should not be duplicate reactions (nor duplicate metabolites).  If duplicates are found, I would recommend aborting and alerting the user that something is wrong with the input.

Reply:
One reaction was duplicated which cause quite an issue, I had to remove it manually. I can add something that checks for that and alerts the user

Reply:
And the duplicate metabolites in that step are fine, that happens if one metabolite that can have more than one mapping is part of the reaction